armi/reactor/tests/test_reactors.py

```diff
@@ -832,6 +832,21 @@ def test_pinCoordsAllBlocks(self):
             coords = b.getPinCoordinates()
             self.assertGreater(len(coords), -1)
 
+    def test_nonUniformAssems(self):
```

Comment:
I love the test! Thanks so much.

But with all the activity on this feature being reverted then put in and then taken out and put in again, I am trying to figure out where the edge cases are in this logic. 

Can we add a more complex case, where there are two or three different items in the "nonUniformAssemsFlags" list?  It should be nearly copy/paste.  OR, if that's a hassle, I can throw in the extra test after this merges.  Thanks!

Reply:
A good place for this would be here
https://github.com/terrapower/armi/blob/908c92d86546d5c6e6e7df946cf1fdfa9515b964/armi/reactor/converters/tests/test_uniformMesh.py#L408

I think a good test would be to make sure that the assembly mesh gets changed to the `refAssem` mesh after `convert` and then after the call to `applyStateToOriginal`, the assembly mesh is back to its original mesh.